pkgs/tools/misc/expect/default.nix

```diff
@@ -28,13 +24,25 @@ tcl.mkTclDerivation rec {
       hash = "sha256-PxQQ9roWgVXUoCMxkXEgu+it26ES/JuzHF6oML/nk54=";
     })
     # Include `sys/ioctl.h` and `util.h` on Darwin, which are required for `ioctl` and `openpty`.
-    ./fix-darwin-clang16.patch
+    # Include `termios.h` on FreeBSD for `openpty`
+    ./fix-darwin-bsd-clang16.patch
+  ] ++ lib.optionals (!stdenv.isFreeBSD) [
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/buildroot/buildroot/c05e6aa361a4049eabd8b21eb64a34899ef83fc7/package/expect/0001-enable-cross-compilation.patch";
+      hash = "sha256-yyzE0Jjac5qaj7Svn4VpMiAqSNLYrw7VZbtFqgMVncs=";
+    })
```

Comment:
So on FreeBSD it cross compiles fine without this patch?  Does that mean we don't need it (and can use the other workarounds) for other platforms too?

Reply:
I doubt this is tenable - the patch hardcodes some linux defaults. It can probably be replaced by moving those defaults into the nix code, but we would need separate defaults for each target system we're cross compiling for.

Reply:
Got it.  I wonder if it should be stdenv.isLinux, then.  (I don't think Darwin->Darwin cross works anyway, but it sounds like this would not be helpful in that situation either…)

Reply:
Can you or someone else with access to a darwin machine test the native build with this diff before I commit it?

Reply:
I don't have easy access.  (I run Linux on a MacBook but Nix does not agree with my macOS installation.)  @Ericson2314 do you?

Reply:
It seems to me like we are making FreeBSD cross compilation of this package more broken than it was before with this one change?

> It can probably be replaced by moving those defaults into the nix code, but we would need separate defaults for each target system we're cross compiling for.

That sounds like a good strategy to me.

Reply:
I think I've fixed it. it looked like there was only one piece which was really bad for FreeBSD, and it required another patch to configure.in because the previous patch bypassed the normal autoconf mechanisms.

Reply:
Is moving the order of the patches intentional / part of the fix?